tools/test_api.py

```diff
@@ -2038,7 +2054,12 @@ def find_tests(base_dir, target_name, toolchain_name, app_config=None):
                 if path_depth == 2:
                     test_group_directory_path, test_case_directory = os.path.split(d)
                     test_group_directory = os.path.basename(test_group_directory_path)
-                    
+
+                    # If the target has no network interface configuration, netsocket tests fail to compile
+                    if not module_config and not configs and \
+                        (test_case_directory == 'netsocket' or test_group_directory == 'netsocket'):
+                        continue
```

Comment:
I remember us going through this at the time, but I think we might be able to solve this in a more general way.

For all of the tests that require the specific macro you use to construct the network interfaces, would it be possible to check for that macro with an `#ifdef` and then use the `#error [NOT_SUPPORTED]` feature of the tools to skip building it?

```c++
#if !MBED_CONF_APP_CONNECT_STATEMENT
    #error [NOT_SUPPORTED] No network configuration found for this target.
#endif
```

Something like that at the top of each test would do the trick.

@sarahmarshy Does that sound right?